langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/OllamaChatModel.java

```diff
@@ -111,14 +117,39 @@ public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecifi
                 .tools(toOllamaTools(toolSpecifications))
                 .build();
 
-        ChatResponse response = withRetry(() -> client.chat(request), maxRetries);
+        ChatModelRequest modelListenerRequest = createModelListenerRequest(request, messages, toolSpecifications);
+        Map<Object, Object> attributes = new ConcurrentHashMap<>();
+        onListenRequest(listeners, modelListenerRequest, attributes);
+
+        try {
+            ChatResponse chatResponse = withRetry(() -> client.chat(request), maxRetries);
+            Response<AiMessage> response = Response.from(
+                    chatResponse.getMessage().getToolCalls() != null ?
+                            AiMessage.from(toToolExecutionRequest(chatResponse.getMessage().getToolCalls())) :
+                            AiMessage.from(chatResponse.getMessage().getContent()),
+                    new TokenUsage(chatResponse.getPromptEvalCount(), chatResponse.getEvalCount())
+            );
+            onListenResponse(listeners, response, modelListenerRequest, attributes);
+
+            return response;
+        } catch (Exception e) {
+            ChatModelErrorContext errorContext = new ChatModelErrorContext(
+                    e,
+                    modelListenerRequest,
+                    null,
+                    attributes
+            );
+
+            listeners.forEach(listener -> {
```

Comment:
Shouldn't it call `OllamaChatModelListenerUtils.onListenError()` here instead?

Reply:
Done.